File: aoj/volume0/n0005/Main.java
import java.util.Scanner;

public class Main {

    public static long gcd(long a, long b) {

        long tmp = a % b;

        while (tmp != 0) {

            a = b;
            b = tmp;
            tmp = a % b;

        }

        return b;

    }

    public static long lcm(long a, long b) {

        return a * b / gcd(a, b);

    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        while (sc.hasNextInt()) {

            long a = sc.nextLong();
            long b = sc.nextLong();

            System.out.println(gcd(a, b) + " " + lcm(a, b));

        }

    }

}

File: aoj/volume0/n0005/Main.cpp
#include <iostream>

using namespace std;

int gcd(int a, int b) {

    int tmp = a % b;

    while (tmp != 0) {

        a = b;
        b = tmp;
        tmp = a % b;

    }

    return b;

}

int lcm(int a, int b) {
    return a / gcd(a, b) * b;
}

int main() {

    int a, b;

    while (cin >> a >> b) {

        cout << gcd(a, b) << " " << lcm(a, b) << endl;

    }

}
